src/Nuget.PackageIndex.VisualStudio/DnxProjectFilter.cs

```diff
@@ -0,0 +1,192 @@
+
+            // Get the DTE
+            var dte = ServiceProvider.GlobalProvider.GetService(typeof(DTE)) as DTE;
+            Debug.Assert(dte != null, "Couldn't get the DTE. Crash incoming.");
+
+            var events = (Events2)dte.Events;
+            if (events != null)
+            {
+                _solutionEvents = events.SolutionEvents;
+                Debug.Assert(_solutionEvents != null, "Cannot get SolutionEvents");
+
+                // clear all cache if solution closed.
+                _solutionEvents.BeforeClosing += OnBeforeSolutionClosed;
+                _solutionEvents.Opened += OnSolutionOpened;
+                _solutionEvents.ProjectRemoved += OnProjectRemoved;
+                _solutionEvents.ProjectRenamed += OnProjectRenamed;
+            }
+        }
+
+        public bool IsProjectSupported(string filePath)
+        {
+            if (string.IsNullOrEmpty(filePath) || _solutionIsClosing)
+            {
+                return false;
+            }
+
+            string projectFullName;
+            if (FilesCache.TryGetValue(filePath, out projectFullName))
+            {
+                return IsProjectFileSupported(projectFullName);
+            }
+
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                // Switch to main thread
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                try
+                {
+                    var container = ServiceProvider.GlobalProvider.GetService<IComponentModel, SComponentModel>();
+                    var hierarchy = DocumentExtensions.GetVsHierarchy(filePath, ServiceProvider.GlobalProvider);
+                    if (hierarchy == null)
+                    {
+                        return;
+                    }
+
+                    var project = hierarchy.GetDTEProject();
+                    if (project == null || string.IsNullOrEmpty(project.FullName))
+                    {
+                        return;
+                    }
+
+                    projectFullName = project.FullName.ToLowerInvariant();
+                }
+                catch (Exception e)
+                {
+                    // we should not throw here, since it would create an exception that may be 
+                    // visible to the user, instead just dump into debugger output or to package
+                    // manager console.
+                    // TODO Package manager console?
+                    Debug.Write(e.ToString());
+                }
+            });
+
+            if (!string.IsNullOrEmpty(projectFullName))
+            {
+                FilesCache.AddOrUpdate(filePath, (k) => projectFullName, (k, v) => projectFullName);
+            }
+
+            return IsProjectFileSupported(projectFullName);
+        }
+
+        private bool IsProjectFileSupported(string projectFile)
+        {
+            return !string.IsNullOrEmpty(projectFile) && projectFile.EndsWith(".xproj");
```

Comment:
Work with bill if you can find a more Stable pattern that doesn't require changing things I many places

-----Original Message-----
From: "Anton Piskunov" <notifications@github.com>
Sent: ‎5/‎6/‎2015 3:53 PM
To: "NuGet/NuGet.PackageIndex" <NuGet.PackageIndex@noreply.github.com>
Cc: "Yishai Galatzer" <yishai_galatzer@yahoo.com>
Subject: Re: [NuGet.PackageIndex] Antonpis/tests (#6)

In src/Nuget.PackageIndex.VisualStudio/DnxProjectFilter.cs:
> +                    // TODO Package manager console?
> +                    Debug.Write(e.ToString());
> +                }
> +            });
> +
> +            if (!string.IsNullOrEmpty(projectFullName))
> +            {
> +                FilesCache.AddOrUpdate(filePath, (k) => projectFullName, (k, v) => projectFullName);
> +            }
> +
> +            return IsProjectFileSupported(projectFullName);
> +        }
> +
> +        private bool IsProjectFileSupported(string projectFile)
> +        {
> +            return !string.IsNullOrEmpty(projectFile) && projectFile.EndsWith(".xproj");
well ..., it is temporary and we use same pattern in other places too, so it should be good enough (if extension changes it will be changed here too)
—
Reply to this email directly or view it on GitHub.